### ISSUE TITLE
Configure ODK Collect to use MapBox as default basemap

### ISSUE THREAD

**Issue by ivangayton** [ivangayton/field-map-odk#2]
# Basemap configuration in ODK Collect

To work well with Select from File (select from map) questions showing OSM data, it's better to have MapBox as the basemap. Google (the default basemap) doesn't match the OSM features (so a building centroid won't be obviously associated with the basemap building polygon) and the OSM layer breaks something. MapBox works. 

The QR Code configuration for ODK Collect appears to reset most parameters. For example, if you configure ODK Collect with a QR code for one form, and modify the settings to use MapBox as the basemap, this only persists until you configure ODK Collect with a new QR code, at which point Google slinks back as the basemap. 

It would be great if we could either:
- Include a setting somewhere in the xlsform that sets the basemap
- Instruct people how to set MapBox as the basemap in a persistent fashion

**Comment by ivangayton** [comment]
Update: looks like this is eminently possible and the relevant documentation is [here](https://docs.getodk.org/collect-import-export/?highlight=configure).

From that link, here are the map source keys:
```
// Maps
    "basemap_source": {"google", "mapbox", "osm", "usgs", "stamen", "carto"},
    "google_map_style": {1, 2, 3, 4},
    "mapbox_map_style": {"mapbox://styles/mapbox/light-v10", "mapbox://styles/mapbox/dark-v10", "mapbox://styles/mapbox/satellite-v9", "mapbox://styles/mapbox/satellite-streets-v11", "mapbox://styles/mapbox/outdoors-v11"},
    "usgs_map_style": {"topographic", "hybrid", "satellite"},
    "carto_map_style": {"positron", "dark_matter"},
    "reference_layer": String, // Absolute path to mbtiles file
```

However, there's a hassle: it looks like this can't be configured in the xlsform, but rather has to be built into the QR code, which probably means we'll have to build our own QR codes.

**Comment by ivangayton** [comment]
Sorted in [this commit](https://github.com/ivangayton/field-map-odk/commit/e4a887c7e3cdb877922d8cd8cc60dcb9993a6788). Relevant ODK documentation [here](https://docs.getodk.org/collect-import-export/); gives explanation of all the keys, and explains how to create a QR code directly using [segno](https://segno.readthedocs.io/en/latest/) (which works rather beautifully).

The ODK documentation isn't terribly clear about how to get the actorIds (we got them by creating a dict of all app users with their displayNames as keys and their actorIds as values) or the roles (we figured out that 2 works to assign an app user to a form by doing so in the web interface and decompiling the resulting QR code).
